src/figuresClasses.ts

```diff
@@ -1,11 +1,87 @@
-export interface Figure {}
+type Shapes = 'circle' | 'triangle' | 'rectangle';
+type Colors = 'red' | 'green ' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle side length must be greater than 0');
+    }
+
+    const biggestSide = [a, b, c].reduce((ac, cu) => (ac > cu ? ac : cu));
 
-export class Circle implements Figure {}
+    const smallestSides = [a, b, c]
+      .filter((num) => num !== biggestSide)
+      .reduce((acc, curr) => curr + acc, 0);
```

Comment:
```suggestion
    const smallestSides = Math.min(...([a, b, c].filter((side) => side !== biggestSide)));
```

Reply:
In this case, my const smallestSides should  contain the result of adding two smalest numbers, so Math.min choose just one of them (